src/west/app/project.py

```diff
@@ -1349,6 +1349,11 @@ def decide_update_strategy(self, project, sha, stats, take_stats):
         if take_stats:
             stats['get current branch HEAD'] = perf_counter() - start
         current_branch = cp.stdout.decode('utf-8').strip()
+        if not current_branch:
+            raise RuntimeError(
```

Comment:
Can you please use `log.die` instead of raising a RuntimeError? Otherwise LGTM, and thank you for the patch!

Reply:
Done 😊